main/src/cgeo/geocaching/calculator/FormulaParser.java

```diff
@@ -0,0 +1,379 @@
+package cgeo.geocaching.calculator;
+
+import cgeo.geocaching.R;
+import cgeo.geocaching.location.Geopoint;
+import cgeo.geocaching.settings.Settings;
+import cgeo.geocaching.utils.CollectionStream;
+import cgeo.geocaching.utils.MatcherWrapper;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+/**
+ * Parse coordinates with formulas.
+ */
+public final class FormulaParser {
```

Comment:
An example formula in the class javadoc would be great

Reply:
Done